### PR TITLE
Swift 4.0 support added

### PR DESCRIPTION
**Pull request by lanserxt**

### CODE REVIEW

**Comment by dcharbonnier** [comment]
thx, can you fix the the travis configuration or the tests please ?

**Comment by dcharbonnier** [comment]
You can add it to this PR or do an other PR

**Comment by lanserxt** [comment]
Will update tests also

**Comment by dcharbonnier** [comment]
great, thanks @lanserxt !

**Comment by lanserxt** [comment]
@dcharbonnier all done, safe to merge

**Comment by lanserxt** [comment]
@gringoireDM What version should I set? Currently set 1.1 for Xcode and 0.10.2 for pod spec.
Swift version file added.

File: HanekeSwift.podspec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Pod::Spec.new do |s|
   s.name = 'HanekeSwift'
   s.module_name = 'Haneke'
-  s.version = '0.10.1'
+  s.version = '0.10.2'
```
Comment (gringoireDM) [inline HanekeSwift.podspec]: The podspec version should be the same of the project version. the issue here is that pod support has been discontinued after 0.10.2. If you wanted latest commits you had to refer to specific branches or straight to master. this is not a best practice and a new pod version should be released. Good 1.1 as project version, but here too it should change to 1.1

To the maintainer of the project: Please tag master after merge and release a pod. This is something we cannot do, so we need you to do it.

**Comment by dcharbonnier** [comment]
No one has access to the pod  account. If you want to release it you will have to change the name

**Comment by gringoireDM** [comment]
@dcharbonnier please do. Haneke_Swift maybe? To release new pod version is important if you want still maintain the project.

**Comment by dcharbonnier** [comment]
Thx !!
For the pod, I don't even have a Mac anymore ;-)

**Comment by gringoireDM** [comment]
Just admit that the library is discontinued. Anyway we moved away from it. Thank you anyway.

**Comment by lanserxt** [comment]
@dcharbonnier You are welcome!

**Comment by dcharbonnier** [comment]
With people like @lanserxt I think we can say it's alive and the more people will contribute the more this project will be alive. 
If there is a better project, with more contributor people or a project you like more people can move away and are welcome back when they want :-) It's like cocoapod vs carthage.
Now if someone want to take the responsibility of deploying and maintaining a pod, he is free to do it and will get my support, this project is open.

**Comment by gringoireDM** [comment]
You don't even have a Mac to test the code you are going to approve. How can you say this is a maintained project?

 I do understand the importance of contributions, but a pod cannot stand just on contributions. It's important that the maintainer helps as well at least with the most basic things like tagging releases, testing before merging a PR, releasing new pod versions. There are things that contributors cannot do. If you don't whish to do them either, handover the ownership of the repository or declare it unmaintained. In any case, like I said before, we already removed haneke as dependency since you are willing to approve anything that pass the travis build. 

We are off topic.

**Comment by smhk** [comment]
Hoy caramba !

Cocoapods is horrible.  Carthage is the only way to go.

one vote!

(Soon, Swift Package Manager will be here and cocoapods will finally be dead.)

**Comment by dcharbonnier** [comment]
[Join the chat at https://gitter.im/Haneke/HanekeSwift](https://gitter.im/Haneke/HanekeSwift?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)

**Pull request merged**